README.md

```diff
@@ -1,3 +1,128 @@
+# Cartoomics Figure Generation 
+
+This Cartoomics repository consists of a workflow written in python that can generate an interaction network using a given knowledge graph from a user derived description of a cartoon image. The goal of this workflow is to generate complete and detailed pathway diagrams using the vast information contained within a knowledge graph.
+
+## Getting Started
+
+These instructions will provide the necessary environments, programs with installation instructions, and input files in order to run the creating_subgraph_using_cosinesim.py script, which will run this workflow from end to end.
+
+The path search algorithms available are:
+
+- Shortest Path Search: find the shortest path between each source and target node (may not be identical each time)
+- Cosine similarity Prioritization: prioritize all shortest paths between each source and target node by maximizing total cosine similarity of the start and all intermediate nodes to the target node in each path
+- Path Degree Product Prioritization: prioritize all shortest paths between each source and target node by maximizing path degree product of all nodes in each path
+
+This ReadMe currently describes the Cosine similarity Prioritization algorithm.
+
+### Prerequisites
+The following software or software packages must be installed, as detailed in the requirements.txt:
+```
+- Python (any version later than 3.6)
+- python-igraph (any version later than 0.9.10)
+- gensim (any version later than 4.2.0)
+- py4cytoscape (any version later than 1.3.0)
+- Cytoscape (any version later than 3.8.0)
+```
+
+Ensure that Cytoscape is open and running on your computer before continuing.
+
+## Running the Script
+
+### Input Directory
+
+The following files must exist in the input directory:
+
+<img width="994" alt="Screen Shot 2022-08-25 at 9 37 40 AM" src="https://user-images.githubusercontent.com/70932395/186709025-6eb4495d-6c12-4353-a1af-5626d835e4ea.png">
+
+### Defaults
+  
+The following values will be used if not otherwise specified:
+- embedding dimensions: embeddings of the knowledge graph will be generated using node2vec of dimension 128, unless otherwise specified
+  --embedding-dimensions <int>
+- weights: edges will not be weighted unless otherwise specified. Any edges specified will be weighted lower/more important in the path search.
+  --weights "[edge_1, edge_2]"
+- search type: the shortest path algorithm used (contained within the python-igraph package) will search for paths in all directions, unless otherwise specified
+  --search-type one
+
+### Command Line Argument 
+  
+To run the script, execute the following command once the input directory is prepared:
+  
+```
+python creating_subgraph_using_cosinesim.py --input-dir INPUTDIR --output-dir OUTPUTDIR
+```
+*Note that the specified output directory must already exist
+ 
+## Expected Outputs
+  
+The creating_subgraph_using_cosinesim.py script will always generate the following files:
```

Comment:
Glad that you pointed this out. Though it may be a nice feature for the program to create the output directory if not already created.

Reply:
agreed, we can add to issues